### PR TITLE
[bazel] Add support for the CW340

### PR DESCRIPTION
**Pull request by pamaury**
TODO, this is a work in progrss. I was able to make it work on fred after fixing the tty devices permissions and running
```
bazel test --test_output=streamed --cache_test_results=no //sw/device/tests:example_test_from_flash_cw340_rom_with_fake_keys --test_arg=--uarts=/dev/ttyUSB2,/dev/ttyACM0
```

### CODE REVIEW

File: rules/opentitan_test.bzl [hunk 1]
```diff
@@ -231,9 +248,6 @@ def cw310_params(
                           baked into the bitstream).
         @param clear_bitstream: Clear FPGA bitstream at the end of the test.
     """
-    if interface not in ("cw310", "hyper310"):
-        fail("The interface must be either 'cw310' or 'hyper310'")
-
```
[the remark this answers is not in the document]
Comment (pamaury) [reply rules/opentitan_test.bzl:236]: This is to ensure that one doesn't for example do `cw310_params(interface="cw340", ...)`: it checks that the interface matches the execution platform and is valid. This avoid obscure errors that occur at runtime.

**Review by engdoreis — APPROVED** [review]
LGTM, thanks @pamaury

File: sw/device/lib/arch/device_fpga_cw340.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/drivers/ibex.h"
+#include "sw/device/silicon_creator/lib/rom_print.h"
+
+/**
+ * @file
+ * @brief Device-specific symbol definitions for the ChipWhisperer CW310 device.
+ */
+
+const device_type_t kDeviceType = kDeviceFpgaCw340;
```
[the remark this answers is not in the document]
Comment (pamaury) [reply sw/device/lib/arch/device_fpga_cw340.c:17]: Yes that's possible, I am just not sure what tests but that can be changed easily after the fact indeed.

**Comment by pamaury** [comment]
This seems to significantly increase the number of targets to build due to the CW340 binaries being different from the CW310. This probably explains the timeout. Maybe, as @engdoreis suggests, it would be better to treat the CW310 and CW340 as identical for now.

**Comment by pamaury** [comment]
It seems that aliasing the two did not work as expected. My next try will be to keep distinct binaries but not build CW340 targets by default. I will then only selectively enable cw340 on some tests.

**Review by jwnrt — APPROVED** [review]
Thanks @pamaury, this LGTM.

I'm no Bazel expert, but the changes look sensible and your commit message explanations make sense. I see the bitstream build takes ~70 minutes but is done in parallel with the CW310 build, so given enough resources this shouldn't affect CI times too much.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**